### ISSUE TITLE
Memory leak/high memory usage

### ISSUE THREAD

**Issue by rbestbmj** [echocat/site24x7_exporter#2]
I started running this exporter on two of our EC2 instances just over a week ago, scraping every 15 seconds, and I have observed that memory usage on the instances has been climbing steadily from an, up until then, consistent ~2.3GB to 5.4GB.

Here's a graph of that growth:
![1](https://user-images.githubusercontent.com/24892471/43890579-105f11d0-9bbf-11e8-9099-3fd8f1ded10a.png)

It seems to have plateaued eventually but even if this isn't strictly a leak, that seems like a lot of memory for this kind of process.
 
Running `top` confirmed that it was the exporter occupying this new slice of memory usage, although I've since lost the output, so I can't share it here.

The memory usage dropped back to pre-exporter levels after restarting the process:
![2](https://user-images.githubusercontent.com/24892471/43890615-215a0c74-9bbf-11e8-9760-8158f6c5908d.png)

but presumably will reoccur again after enough time. The fact that it happened on two instances leads me to believe it isn't just a freak occurrence.

I'm running this version of the exporter:
https://github.com/echocat/site24x7_exporter/releases/download/v0.1.6/site24x7_exporter-linux-amd64

Some system information:
```
Linux 3.13.0-74-generic #118-Ubuntu SMP Thu Dec 17 22:52:10 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.3 LTS
Release:	14.04
Codename:	trusty
```

I haven't had the time to dive into the code and try and figure out if there are any potential leaks, do you have any ideas?

**Comment by rbestbmj** [comment]
@blaubaer @joshk0

**Comment by joshk0** [comment]
If you open up a [pprof port](https://artem.krylysov.com/blog/2017/03/13/profiling-and-optimizing-go-web-applications/), you would be able to check for the allocations that are taking up space while you see the issue occurring.

**Comment by rbestbmj** [comment]
Thanks for the hint @joshk0! I'll do that and let you know what I find.

**Comment by blaubaer** [comment]
Hi guys! Sorry for reply that late. I really have issues to reproduce this issue because I not have an account there anymore. :-( I guess someone else has to identify the issue in the code that I can fix it. Sorry!

**Comment by mdiazv** [comment]
I guess this issue got fixed with #3

**Comment by blaubaer** [comment]
I think so, too.
